### PR TITLE
fix(INJI-262):pinned VC's audit logs are missing

### PR DESCRIPTION
**Pull request by srikanth716**
https://github.com/tw-mosip/inji/assets/97477121/0f05104a-06b2-44ad-a58f-aff1c341b4c8

### CODE REVIEW

File: machines/vcItem.ts [hunk 1]
```diff
@@ -782,6 +782,20 @@ export const vcItemMachine =
...
 
+        logVcPinToggle: send(
+          (context) =>
+            ActivityLogEvents.LOG_ACTIVITY({
+              _vcKey: VC_ITEM_STORE_KEY(context),
+              type: context.isPinned ? 'VC_PIN' : 'VC_UNPIN',
+              timestamp: Date.now(),
+              deviceName: '',
+              vcLabel: context.id,
+            }),
```
Comment (vharsh) [inline machines/vcItem.ts:793]: IMO, the ticket states that the **audit logs** of the specifically selected VC should be shown and not the action of logging the VC being pinned & unpinned. Am I missing a discussion, if so please update the Jira ticket with the newer description.

**Comment by vharsh** [comment]
@srikanth716 Please attach screenshots here and in Jira ticket

File: screens/Home/MyVcs/HistoryTab.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export const HistoryTab: React.FC<HistoryTabProps> = (props) => {
         onDismiss={controller.DISMISS}>
         <Column fill>
           {controller.activities.map((activity) => {
-            if (activity._vcKey == props.vcKey) {
+            if (activity._vcKey.split(':')[3] == props.vcKey.split(':')[3]) {
```
Comment (swatigoel) [inline screens/Home/MyVcs/HistoryTab.tsx:33]: extract it as variable and the use it for readibility.

**Review by swatigoel — CHANGES_REQUESTED** [review]

**Comment by swatigoel** [comment]
We could directly revert the commit `31b88dab1e5ad25ff0f139a456e562a2d1101ed8` to avoid any new bug because of changes.
We can take care of this in future

File: shared/constants.ts [hunk 3]
```diff
@@ -21,6 +22,11 @@ export const VC_ITEM_STORE_KEY = (vc: Partial<VC>) =>
 export const VC_ITEM_STORE_KEY_REGEX =
   '^vc:(UIN|VID):[0-9]+:[a-z0-9-]+:[true|false]+$';
 
+//To compare the vckey with requestId, when the vc is pinned
+export const isVcKeyMatch = (vckey: string, CompareVckey: string) => {
```
Comment (swatigoel) [inline shared/constants.ts:26]:
```
export const isSameVC = (vcKey: string, pinnedVcKey: string) => {
const requestId = vcKey.split(':')[3];
const pinnedRequestId = pinnedVcKey.split(':')[3];

return requestId === pinnedRequestId;
}
```

File: screens/Home/MyVcs/HistoryTab.tsx [hunk 4]
```diff
@@ -30,7 +31,8 @@ export const HistoryTab: React.FC<HistoryTabProps> = (props) => {
         onDismiss={controller.DISMISS}>
         <Column fill>
           {controller.activities.map((activity) => {
-            if (activity._vcKey == props.vcKey) {
+            const vcKeyMatch = isVcKeyMatch(activity._vcKey, props.vcKey);
```
Comment (swatigoel) [inline screens/Home/MyVcs/HistoryTab.tsx:34]: isSameVC(activity._vcKey, props.vcKey)

**Review by swatigoel — CHANGES_REQUESTED** [review]

**Pull request merged**